packages/flutter/test/material/buttons_test.dart

```diff
@@ -1195,4 +1196,123 @@ void main() {
     final Material material = tester.widget<Material>(rawButtonMaterial);
     expect(material.color, const Color(0xff00ff00));
   });
+
+  testWidgets('MaterialButton should be enabled when onLongPress is not null.', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: MaterialButton(
+          onPressed: null,
+          onLongPress: () { /* to make sure the button is enabled */ },
+          child: const Text('button'),
+        ),
+      ),
+    );
+
+    final MaterialButton button = tester.widget<MaterialButton>(find.byType(MaterialButton));
+    expect(button.enabled, true);
+  });
+
+  testWidgets('FlatButton should be enabled when onLongPress is not null.', (WidgetTester tester) async {
```

Comment:
I just realized that `buttons_test.dart` is an odd file and contains an amalgamation of tests for the different types of buttons, even though `flat_button_test.dart`, `raised_button_test.dart`, and `material_button_test.dart` exist. 

It could be nice to have each button widget's tests in their own files, but maybe that's out of the scope of this PR since there are preexisting tests in this file that test for the different buttons as well.

Reply:
Oh, you are right. For some reason I thought everything was on the same (because, well, there are everything on the same). I'm going to split those on the correct files.

Just the same question as I commented below. How many tests. 2 (onPressed true + onLongPress false / onPressed false + onLongPress true) or 4 (onPressed false + onLongPress false.. and so on)?

Reply:
I think the following combinations will be important:
1) just onPressed to make sure that functionality works
2) just onLongPress to make sure that functionality works
3) both onPressed and onLongPress activated, then testing between different the two gestures to ensure that both are properly and distinctly recognized
4) both deactivated to ensure that neither callbacks are being invoked when they shouldn't be

Definitely check the other test files to see if some of this is already implemented -- I'd be surprised if the onPressed ones weren't already tested, but you never know :)

Reply:
So.. outlineButton contained onTap test. I modified it to include onLongPress and replicated into the others. RawMaterialButton already contained one test, I kept it, but added the new one, more complex, in the end of the file. The others had no onTap/onPressed test.

Reply:
Also, running `flutter analyze --flutter-repo` will run the analyzer locally so you do not have to wait for the CI to finish running its tests

Reply:
I had the issue where the repo version I was using was newer than flutter I had, therefore that wouldn't work. I needed to start using ../../bin/flutter instead. Wasnt trivial. And that doesn't check the whitespace thing.